lib/core.js

```diff
@@ -510,6 +511,8 @@ exports = module.exports = internals.Core = class {
 
             const request = Request.generate(this.root, req, res, options);
 
+            Channels.onRequestChannel.publish(request);
```

Comment:
If this assignment is in the `request._execute()`, perhaps it is to avoid executing it in case of heavy load ?

I would vote to keep the `publish()` here given we need another event when a request is routed. I could add an event early in the `request._lifecycle()` method ? Something like `hapi.onRequestLifecycle` ?